### DISCUSSION TITLE
Cover gallery?

### DISCUSSION THREAD

**Discussion by saltmanz** [yegon666/CbzMage#6] [Ideas]
Just a thought, but it would be neat to have the option to define a folder in which to save the full-size cover image of every converted comic.

**Comment by ToofDerling** [comment]
That would certainly be possible. What do you think of this: saving the covers in the CbzDir structure? 

Like this CbzDir/Publisher/Title.jpg

**Reply by saltmanz** [reply]
I'd want them in separate folders from the CBZs. (The comics themselves are saved as CbzDir/Publisher/Title.cbz, correct?)

I think ideally I'd just want a 3rd "main" folder to alongside the scans and the titles.

**Reply by ToofDerling** [reply]
Okay, that's doable. I think I'll implement both, cover images alongside the cbz or in a separate dir. Should cover images in their own directory be sorted by publisher (like "[Marvel] Young Avengers Vol. 1 Style Substance.jpg") or should it just be the title? 

If it's just the title there's a small (very small I guess) chance that covers from titles with the same name but from different publishers will overwrite each other.

**Reply by saltmanz** [reply]
I don't see a problem with including the publisher in the filename.

**Comment by ToofDerling** [comment]
v0.6 is up. If SaveCover is true and SaveCoverDir is specified cover images will be saved there.

**Reply by saltmanz** [reply]
I like it! Not sure I need the "NEW" in front of the filename, though. And I wish it were possible to regenerate the cover without re-converting the book. But those are very minor complaints.

**Reply by ToofDerling** [reply]
The ".New" thing is a bug and easy to fix, I'll do that right away.

**Comment by ToofDerling** [comment]
v0.7 is up with a fix for the saved cover filename

**Comment by ToofDerling** [comment]
v0.8 is up with a SaveCoverOnly option. Usable for regenerating covers.

It's a bit slower then I expected, but obviously much faster then a full convert. I saved 1900 covers in about 12 minutes.
